src/main/scala/io/iohk/ethereum/vm/VM.scala

```diff
@@ -36,7 +36,7 @@ class VM extends Logger {
       case Some(opCode) =>
         val newState = opCode.execute(state)
         import newState._
-        log.trace(s"$opCode | pc: $pc | depth: ${env.callDepth} | gas: $gas | stack: $stack")
+        log.trace("{} | pc: {} | depth: {} | gas: {} | stack: {}", opCode, Integer.valueOf(pc), Integer.valueOf(env.callDepth), gas, stack)
```

Comment:
So this is fine, but while we don't have `scala-logging` how about we do what it does under the hood, meaning:

```
        if (log.isTraceEnabled) {
          log.trace(s"$opCode | pc: $pc | depth: ${env.callDepth} | gas: $gas | stack: $stack")
        }
```

That would be even more performant than the proposed solution.

Reply:
yep, it will avoid the cost of array allocation